### PR TITLE
Rubocop linting and performance violations

### PR DESCRIPTION
**Pull request by exbinary**
## Why
Building on #935, fixes violations of `Lint/*` and `Performance/*` rules.

### What
See individual commits for the list of rule violations addressed. Diffs probably also make more sense one commit at a time.

In many instances, i left `rubocop:todo`s instead of fixing the violation (this is like `rubocop:disable` but reveals that we really want to fix the violation instead of disable it).

See inline PR comments for specific questions.

### Testing
Relying on our existing suite to catch any regressions.

### Next Steps
?

### Outstanding Questions, Concerns and Other Notes
?

### Pre-Merge Checklist
- [x] Security & accessibility have been considered
- [x] Tests have been added, or an explanation has been given why the features cannot be tested
- [x] Documentation and comments have been added to the codebase where required
- [ ] Entry added to CHANGELOG.md if appropriate
- [ ] Outstanding questions and concerns have been resolved
- [ ] Any next steps have been turned into Issues or Discussions as appropriate

### CODE REVIEW

File: app/controllers/contributions_controller.rb [hunk 1]
```diff
@@ -52,9 +52,9 @@ def contribution_blueprint_options
   end
 
   def filter_params
-    return {} unless allowed_params&.to_h.any?
+    return {} unless allowed_params.to_h.any?
```
Comment (exbinary) [inline app/controllers/contributions_controller.rb:55]: I don't think `allowed_params` can be nil?

File: app/forms/base_form.rb [hunk 2]
```diff
@@ -20,20 +20,6 @@ def self.filter_keys(source = self)
...
-    # FIXME: ActiveInteraction::Base#given? seems to have a bug recognizing multipart date params
-    raw_inputs = @_interaction_inputs
-    inputs.select do |key|
-      given?(key) || (
-        raw_inputs.key?(:"#{key}(1i)") &&
-        raw_inputs.key?(:"#{key}(2i)") &&
-        raw_inputs.key?(:"#{key}(3i)")
-      )
-    end
-  end
```
Comment (exbinary) [inline app/forms/base_form.rb:33]: Moved to the end of the file.

File: config/initializers/simple_form_bulma.rb [hunk 3]
```diff
@@ -543,56 +543,56 @@
     # end
 
 # rubocop:enable Layout/CommentIndentation
+end
```
Comment (exbinary) [inline config/initializers/simple_form_bulma.rb:546]: Everything below here in this file is just whitespace change; moved the class definitions below out of the block above.

File: db/migrate/20200523184552_add_auto_generated_to_communication_log.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 class AddAutoGeneratedToCommunicationLog < ActiveRecord::Migration[6.0]
   def up
     add_column :communication_logs, :auto_generated, :boolean
-    change_column_default :communication_logs, :auto_generated, :false
+    change_column_default :communication_logs, :auto_generated, false
```
Comment (exbinary) [inline db/migrate/20200523184552_add_auto_generated_to_communication_log.rb:4]: Fixing but `:false` seems to set the default correctly.

File: db/scripts/submission_response_seeds_gitignored_csvs.rb [hunk 5]
```diff
@@ -7,13 +7,14 @@
 importer = Importers::SubmissionResponseImporter.new(User.first, csv)
 importer.import(path)
 
+# TODO process offers and old_offers as well
```
Comment (exbinary) [inline db/scripts/submission_response_seeds_gitignored_csvs.rb:10]: I've tried to interpret the meaning of the pre-existing comment here.

File: app/controllers/contributions_controller.rb (hunk 1 above)
Comment (h-m-m) [reply app/controllers/contributions_controller.rb:55]: Right. It can be empty/blank or it can raise an error, but it can't be `nil`

**Comment by exbinary** [comment]
Reviewed in person with @svileshina .

**Pull request merged**